### ISSUE TITLE
Add Range method with step

### ISSUE THREAD

**Issue by AlexRadch** [morelinq/MoreLINQ#827]
The Enumerable.Range does not have step parameter. Can you add Range method with step parameter, please. `Range(int start, int count, int step)`

It is different from Sequence(int start, int stop, int step) that Range will use count parameter instead stop boundary.

It will cover cases when you want get needed quantity of numbers with needed step between them.

It is like Sequence(start, start + step * (count - 1), step)

Also it will not generate infinite sequence if step is zero.

**Comment by AlexRadch** [comment]
```cs
        public static IEnumerable<int> Range(int start, int count, int step) {
            long max = ((long)start) + count - 1;
            if (count < 0 || max > Int32.MaxValue) throw Error.ArgumentOutOfRange("count");
            return RangeIterator(start, count, int step);
        }

        static IEnumerable<int> RangeIterator(int start, int count, int step) {
            var value = start;
            for (int i = 0; i < count; i++) 
            {
                yield return value;
                value += step;
            }
        }
```

**Comment by viceroypenguin** [comment]
@AlexRadch FYI: this overload has been included in SuperLinq 4.0.0, which is now released.

**Comment by atifaziz** [comment]
You can achieve this with MoreLINQ's `Generate` combined with LINQ's `Take` or roll it up into a simple extension method if you need to do this often:

```c#
static class Seq
{
    public static IEnumerable<int> Step(int start, int step, int count) =>
        MoreEnumerable.Generate(start, x => x + step).Take(count);
}
```

I am going to close this on the grounds that it's not adding anything new that cannot be achieved pretty trivially with a one-liner.
